src/settings/global_user.rs

```diff
@@ -8,11 +8,11 @@ use serde::{Deserialize, Serialize};
 use crate::terminal::emoji;
 use config::{Config, Environment, File};
 
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct GlobalUser {
-    pub email: Option<String>,
-    pub api_key: Option<String>,
-    pub api_token: Option<String>,
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub enum GlobalUser {
+    TokenAuthUser { api_token: String },
+    KeyAuthUser { email: String, api_key: String },
```

Comment:
nit here: can we make this `TokenAuth` and `GlobalKeyAuth` - I feel like `User` is redundant here, and also `Token` and `Key` are very similar

Reply:
+1